### PR TITLE
272006 - Add dog details page

### PR DESCRIPTION
**Pull request by sfitz42**
- Add dog details page
- Add expiry date calc and tests
- Change dog name to required

### CODE REVIEW

File: app/models/cdo/create/dog-details.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      name: 'breed',
+      fieldset: {
+        legend: {
+          text: 'Breed type'
+        }
+      },
+      value: dogDetails[keys.breed],
+      items: breedTypes.map(type => ({
+        text: type.breed,
+        value: type.breed
```
Comment (jbarnsley10) [inline app/models/cdo/create/dog-details.js:19]: Should we not be using breed id as the value?

File: app/views/cdo/create/dog-details.njk [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+{% extends '_layout.njk' %}
+
+{% block pageTitle %}{{ serviceName }}{% endblock %}
+
+{% block beforeContent %}
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-one-half">
+      {{ govukBackLink({
+        text: "Back",
+        href: "/",
```
[the remark this answers is not in the document]
Comment (sfitz42) [reply app/views/cdo/create/dog-details.njk:18]: Just a placeholder for now. Will need to be updated in the owner summary page ticket.

File: app/models/cdo/create/dog-details.js (hunk 1 above)
Comment (sfitz42) [reply app/models/cdo/create/dog-details.js:19]: I'd prefer to have the API do the mapping. Just gives us a bit more flexibility if we decide to change storage methods in the future.

Comment (jbarnsley10) [reply app/models/cdo/create/dog-details.js:19]: Ok, it's a bit of a dilemma since using id's gives other benefits. Happy to leave as is

**Pull request merged**